app/assets/javascripts/campaign_page.js

```diff
@@ -1,60 +1,53 @@
-
-  // This controls adding new checkboxes to the petition form when it is present.
-  var checkbox_count = 0;
-  // We bind the event to the widget location div, then indicate that it should
-  // only fire when the add-checkbox ID is clicked. This is because the add-checkbox
-  // element doesn't exist on page load, but we're trying to create a handler for it already.
-  widget_location.on('click', '#add_checkbox', function(event){
-    event.preventDefault();
-    var checkbox_html = $('#checkbox_seed').html();
-
-    // This replaces the placeholder 'cb_number' section in the checkbox we're pulling out
-    // with the current inline number to be used for the checkbox. So for instance, the first
-    // checkbox which is placed on the page will be in [widget][checkboxes][0][label],
-    // while the second will be in [widget][checkboxes][1][label] to differentiate them
-    // when they're saved in the database.
-    var final_html = checkbox_html.replace(/{cb_number}/g, checkbox_count);
-    $('#checkbox_container').append(final_html);
-    checkbox_count++;
-  });
-
-  widget_location.on('click', '#add_textarea', function(event){
-    event.preventDefault();
-    var textarea_html = $('#textarea_seed').html();
-    var final_html = textarea_html.replace('[placeholder]', '');
-    $('#textarea_container').html(final_html);
-  });
-
-  $('#action-button').on('click', function(event){
-    //Don't actually submit anything
-    event.preventDefault();
-    $.ajax({
-      url: '/campaign_pages/sign/',
-      method: 'post',
-      success: function(data, textStatus, jqXHR) {
-        var redirect_location = $('#redirect-location').html();
-        if(redirect_location != '') {
-          window.location.href = redirect_location;
-        }
+(function(){
+
+  var configureDropZone = function() {
+    Dropzone.options.dropzone = {
+      maxFilesize: 2,
+      paramName: "image[content]",
+      addRemoveLinks: false,
+      previewsContainer: null,
+      createImageThumbnails: true,
+
+      init: function() {
+        this.on("success", function(resp, data) {
+          $('.campaign-images').append(data.html);
+        });
+
+        this.on("addedfiled", function(file) {
+          this.removeFile(file);
+        });
       }
-    });
-  });
+    };
+  }
+
+  var configureQuillEditor = function() {
+    var quillConfig = {
+      theme: 'snow',
+      modules: {
+        'toolbar': { container: '#toolbar' },
+        'link-tooltip': true
+      }
+    },
+
+    quill = new Quill('#editor', quillConfig),
+    $contentField = $('#campaign_page_content'),
+
+    updateContentBeforeSave = function(){
+      var content = quill.getHTML();
+      $contentField.val(content);
+    };
+
+    quill.setHTML( $contentField.val() );
+
+    $('form.edit_campaign_page').on('ajax:before', updateContentBeforeSave);
+  }
+
+  var initialize = function() {
+    configureDropZone();
+    configureQuillEditor();
+  };
```

Comment:
One note -- at some point (probably post launch) it would be great to have
track changes/authorship/revert/comment functionality over some or all of
the campaign page components -- so that we could see changes and do editing
of campaigns in Champaign rather than in Google Docs as we currently do.
Not sure if their are WYSIWYG tools that already have that build in.

On Wed, Aug 12, 2015 at 9:05 AM, Omar Sahyoun <notifications@github.com>
wrote:

> In app/assets/javascripts/campaign_page.js
> <https://github.com/SumOfUs/Champaign/pull/90#discussion_r36856569>:
>
> > +    $contentField = $('#campaign_page_content'),
> > +
> > +    updateContentBeforeSave = function(){
> > +      var content = quill.getHTML();
> > +      $contentField.val(content);
> > +    };
> > +
> > +    quill.setHTML( $contentField.val() );
> > +
> > +    $('form.edit_campaign_page').on('ajax:before', updateContentBeforeSave);
> > +  }
> > +
> > +  var initialize = function() {
> > +    configureDropZone();
> > +    configureQuillEditor();
> > +  };
>
> configureDropZone - set up lib I'm using for async image uploads.
> configureQuillEditor - set up the simple quill editor for page content.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SumOfUs/Champaign/pull/90/files#r36856569>.
>



-- 
*Paul Ferris*
Campaign Director | SumOfUs.org <http://sumofus.org>
+ 1 917 753 9204 (New York)
skype: p.ferris